src/main/java/net/cardosi/mojo/TestMojo.java

```diff
@@ -15,301 +15,443 @@
+        } finally {
+          if (driver != null) {
+            driver.quit();
+          }
         }
+      }
+    } catch (ProjectBuildingException | IOException e) {
+      throw new MojoExecutionException("Failed to build project structure",
+                                       e);
     }
 
-
-    private static boolean isSuccess(WebDriver d) {
-        return (Boolean) ((JavascriptExecutor) d).executeScript("return window.G_testRunner.isSuccess()");
+    if (failedTests.isEmpty()) {
+      getLog().info("All tests were passed successfully!");
+    } else {
+      failedTests.forEach((name, startupHtmlFile) -> getLog().error(String.format("Test %s failed, please try manually %s",
+                                                                                  name,
+                                                                                  startupHtmlFile)));
+      throw new MojoFailureException("At least one test failed");
     }
-
-    private static boolean isFinished(WebDriver d) {
-        return (Boolean) ((JavascriptExecutor) d).executeScript("return !!(window.G_testRunner && window.G_testRunner.isFinished())");
+  }
+
+  private void analyzeLog(WebDriver driver) {
+    logMessageFromDriver(LogType.BROWSER,
+                         driver);
+    if (driver.getPageSource()
+              .contains("closureTestRunnerLog")) {
+      this.logMessageFromDriverHTML(driver.getPageSource());
     }
-
-    /**
-     * If specific tests are specified, will use them, otherwise will look for the tests through includes/excludes,
-     * and produce a build config each.
-     */
-    public static List<ClosureBuildConfiguration> getTestConfigs(ClosureBuildConfiguration baseConfig, List<String> tests, MavenProject project, List<String> includes, List<String> excludes) {
-        List<String> testEntrypoints;
-        if (tests == null || tests.isEmpty()) {
-            testEntrypoints = project.getTestCompileSourceRoots().stream()
-                    .flatMap(s -> getTestEntrypoints(new File(s), includes, excludes).stream())
-                    .distinct()
-                    .map(f -> f.replaceAll("\\.java$", "").replaceAll("/", "."))
-                    .collect(Collectors.toList());
-        } else {
-            testEntrypoints = tests;
-        }
-        return testEntrypoints.stream().map(name -> new TestConfig(name, baseConfig)).collect(Collectors.toList());
+  }
+
+  private void logMessageFromDriver(String logType,
+                                    WebDriver driver) {
+    if (!Objects.isNull(driver)) {
+      driver.manage()
+            .logs()
+            .get(logType)
+            .getAll()
+            .forEach(l -> {
+              if (Level.SEVERE.equals(l.getLevel())) {
+                getLog().error(String.format(stripText(l.getMessage())));
+              } else {
+                getLog().info(String.format(stripText(l.getMessage())));
```

Comment:
yes, you are right. It spams a lot. 
The problem why this spamming is logged, is, that the error message does not know the failling method. I take the output from the browsers console and this will always print INFO:
```
[INFO] 21:07:55.876 : Running test: testExec_verticalTab
[INFO] Error: junit.framework.ComparisonFailure: Wrong group 0
    at ComparisonFailure.m_createError__java_lang_String_$pp_java_lang (eval at <Canonymous> (file:///Users/hoss/Documents/Development/OpenSource/frankhossfeld/GWT-Project-Modules/gwt-regexp-parent/j2cl-tests/target/j2cl-tests-1.0-SNAPSHOT-test/j2c-org.gwtproject.regexp.shared.RegExpJ2clTest.js:1223:3), <Canonymous>:239:12)
    at ComparisonFailure.m_initializeBackingError___$p_java_lang_Throwable (eval at <Canonymous> (file:///Users/hoss/Documents/Development/OpenSource/frankhossfeld/GWT-Project-Modules/gwt-regexp-parent/j2cl-tests/target/j2cl-tests-1.0-SNAPSHOT-test/j2c-org.gwtproject.regexp.shared.RegExpJ2clTest.js:1223:3), <Canonymous>:229:136)
```` 
Otherwise we need a lot more logic to interpret the output and remove not needed logs.

Reply:
change to spam only in case of error